
Fix build when CONFIG_UIDGID_STRICT_TYPE_CHECKS is set.

Signed-off-by: Thomas Backlund <tmb@mageia.org>

--- nvidia-current-331.38/NVIDIA-Linux-x86_64-331.38/kernel/uvm/nvidia_uvm_linux.h.orig	2014-01-09 04:49:00.000000000 +0200
+++ nvidia-current-331.38/NVIDIA-Linux-x86_64-331.38/kernel/uvm/nvidia_uvm_linux.h	2014-01-14 00:59:12.294414999 +0200
@@ -405,12 +405,16 @@ typedef void irqreturn_t;
 // not require the RCU's read lock on current->cred.
 //
 //
+#ifdef CONFIG_UIDGID_STRICT_TYPE_CHECKS
+#define NV_CURRENT_EUID() __kuid_val(current_euid())
+#else
 #if defined(NV_TASK_STRUCT_HAS_CRED)
 #define NV_CURRENT_EUID() \
     (((typeof(*current->cred) __force __kernel *)current->cred)->euid)
 #else
 #define NV_CURRENT_EUID() (current->euid)
 #endif
+#endif
 
 #define NV_ATOMIC_SET(data,val)         atomic_set(&(data), (val))
 #define NV_ATOMIC_INC(data)             atomic_inc(&(data))
